tests/test_multipart.py

```diff
@@ -574,6 +589,23 @@ def test_read_chunk_doesnt_breaks_reader(self):
             while not part.at_eof():
                 yield from part.read_chunk(3)
 
+    def test_read_chunk_from_stream_doesnt_breaks_reader(self):
+        reader = aiohttp.multipart.MultipartReader(
+            {CONTENT_TYPE: 'multipart/related;boundary=":"'},
+            Stream(b'--:\r\n'
+                   b'\r\n'
+                   b'test'
+                   b'\r\n--:\r\n'
+                   b'\r\n'
+                   b'passed'
+                   b'\r\n--:--'))
+        while True:
+            part = yield from reader.next()
+            if part is None:
+                break
+            while not part.at_eof():
+                yield from part.read_chunk(4)
```

Comment:
In your case we are reading by four bytes:
```
BODY\r\n--boundary
```
First chunk will be `BODY`, and `\r\n--` is buffered in `prev_chunk`. Boundary not found yet.
In next call we'll detect end of part, but we couldn't return anything but `b''`, because all preceding content is already returned.
It's possible to raise chunk_size limit to
```python
len(boundary) + len(b'\r\n--')
```
But profit will be much bigger if we don't scan for boundary every chunk twice (as left and right parts of `window`). I'll think about it later.